### ISSUE TITLE
bindings_python_pyiree_compiler_compiler_test failing

### ISSUE THREAD

**Issue by marbre** [google/iree#1166]
Working to get the python bindings build with CMake and running the tests in CI with ctest, I noticed that `bindings_python_pyiree_compiler_compiler_test` is failing. I assume there is a relation to commit 92566df. With PR #1164 merged, the bindings can be build, but the test fails with:
```
+ /usr/bin/python3 /home/mbrehler/repos/iree/bindings/python/pyiree/compiler/compiler_test.py
Traceback (most recent call last):
  File "/home/mbrehler/repos/iree/bindings/python/pyiree/compiler/compiler_test.py", line 17, in <module>
    from pyiree import compiler
  File "/home/mbrehler/repos/iree/build/bindings/python/pyiree/compiler/__init__.py", line 21, in <module>
    from . import binding as binding
ImportError: arg(): could not convert default argument 'value: absl::nullopt_t' in function 'add_option' into a Python object (type not registered yet?)
```

**Comment by marbre** [comment]
@stellaraccident I would be awesome if you can take a look.

**Comment by stellaraccident** [comment]
Sorry about this - this is a classic symptom of older compilers choking on templated stuff (the optional/nullopt bits have come up previously). Let me see if I can work around it (I think I found an incantation earlier that fixes it). What compiler are you using?

**Comment by marbre** [comment]
I tested on a VM with `clang version 8.0.0-3~ubuntu18.04.2`.

**Comment by stellaraccident** [comment]
Since I can't repro this locally, if you have an ability to repro, could you see if changing this in compiler.cc fixes your issue (the key is the absl::nullopt -> absl::optional change on the last line):

```
      "add_option",
      [](std::string name, absl::optional<std::string> value) {
        auto options_map = llvm::cl::getRegisteredOptions();
        auto found_it = options_map.find(name);
        if (found_it == options_map.end()) {
          std::string message = "Unknown LLVM option: ";
          message.append(name);
          throw RaiseValueError(message.c_str());
        }

        std::string value_sr = value ? *value : "";
        found_it->getValue()->addOccurrence(1, name, value_sr);
      },
      py::arg("name"), py::arg("value") = absl::optional<std::string>());
```

**Comment by marbre** [comment]
That fixes the issue! Sidenote: This also occurred with clang9.

**Comment by stellaraccident** [comment]
I'll send a quick fix.

**Comment by marbre** [comment]
Solved with commit b00d64b. Thanks for the quick fix @stellaraccident!
